lib/proxy/peer/dial/dial.go

```diff
@@ -0,0 +1,38 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package peerdial
+
+import (
+	"net"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// Dialer dials a node in the given cluster.
+type Dialer interface {
+	Dial(clusterName string, request DialParams) (net.Conn, error)
+}
+
+type DialParams struct {
+	From     *utils.NetAddr
+	To       *utils.NetAddr
+	ServerID string
+	ConnType types.TunnelType
+}
```

Comment:
Doesn't work because of package import cycles, even getting rid of the `reversetunnelclient.Server` in param position you're still going to have a `GetSite` that returns an object with a `Dial` method that takes a `reversetunnelclient.DialParams` - and requiring the caller to do three layers of rewrapping is almost the entire point of this function, afaict.

Reply:
Thanks for the explanation.